packages/block-library/src/template-part/edit.js

```diff
@@ -1,3 +1,118 @@
-export default function TemplatePartEdit() {
-	return 'Template Part Placeholder';
+/**
+ * WordPress dependencies
+ */
+import {
+	useEntityProp,
+	__experimentalUseEntitySaving,
+	EntityProvider,
+} from '@wordpress/core-data';
+import { useMemo, useCallback } from '@wordpress/element';
+import { parse } from '@wordpress/blocks';
+import { serializeBlocks } from '@wordpress/editor';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+
+const saveProps = [ 'content', 'status' ];
+function TemplatePart() {
+	const [ content, _setContent ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'content'
+	);
+	const [ status, setStatus ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'status'
+	);
+	const initialBlocks = useMemo( () => {
+		if ( status !== 'publish' ) {
+			// Publish if still an auto-draft.
+			setStatus( 'publish' );
+		}
+		if ( typeof content !== 'function' ) {
+			const parsedContent = parse( content );
+			return parsedContent.length ? parsedContent : undefined;
+		}
+	}, [] );
+	const [ blocks = initialBlocks, setBlocks ] = useEntityProp(
+		'postType',
+		'wp_template_part',
+		'blocks'
+	);
+	const [ isDirty, isSaving, save ] = __experimentalUseEntitySaving(
+		'postType',
+		'wp_template_part',
+		saveProps
+	);
+	const saveContent = useCallback( () => {
+		_setContent( content( { blocks } ) );
```

Comment:
In the above `useMemo`, we test whether `content` is a function. Here, we have no such safety checks. Do we run any risk that this isn't a function?

Aside: I'm not really clear _why_ this is a function.

Reply:
`useMemo` runs on the first render. `InnerBlocks` will set it to a function after that.

>Aside: I'm not really clear why this is a function.

It mirrors how the editor store works. It uses a function that computes the eventual value to make the entity dirty in the store without having to run a potentially expensive computation.

Reply:
I'm refactoring this into a custom hook with docs, because it will be used often, `useSyncedEntityInnerBlocks`.